x/generators/serde_graphql_test.go

```diff
@@ -0,0 +1,82 @@
+package generators
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/widmogrod/mkunion/x/shape"
+	"testing"
+)
+
+func TestSerdeGraphQLTagged_Generate_Struct(t *testing.T) {
+	s := &shape.StructLike{
+		Name:          "TestStruct",
+		PkgName:       "test",
+		PkgImportName: "github.com/test/test",
+		Fields: []*shape.FieldLike{
+			{
+				Name: "Name",
+				Type: &shape.PrimitiveLike{Kind: &shape.StringLike{}},
+			},
+			{
+				Name: "Value",
+				Type: &shape.PrimitiveLike{Kind: &shape.NumberLike{}},
+			},
+		},
+	}
+
+	generator := NewSerdeGraphQLTagged(s)
+	result, err := generator.Generate()
+
+	assert.NoError(t, err)
+	assert.Contains(t, result, "GraphQL Schema for TestStruct")
+	assert.Contains(t, result, "type TestStruct")
+	assert.Contains(t, result, "Name: String!")
+	assert.Contains(t, result, "Value: Float!")
+	assert.Contains(t, result, "Example GraphQL Resolver")
```

Comment:
issue: partial occurence does not fullfil educationa value. Engineer who is seeign this code for the first time needs to build intutition wht this code does, and what is responsible for. Tests should help with it. First test demonstrating concept should assume that user does not know what is expected behaviour or givent input what output should look like.

To improve this and other tests instead of partial asserts, assert whole output structure. That way it will act as documentation and demonstration of what to expect from function.